farm/contract/lib.rs

```diff
@@ -272,6 +272,18 @@ mod farm {
             Ok(())
         }
 
+        fn reward_rates_to_u128(&self) -> Vec<u128> {
+            self.farm_reward_rates
+                .iter()
+                .map(|rr| {
+                    rr.0.checked_div(U256::from(SCALING_FACTOR))
+                        .unwrap()
+                        .try_into()
+                        .unwrap()
```

Comment:
This shouldn't ever fail, but I think we probably still shouldn't make plain unwraps

Reply:
You're right. I'll emit the `MathError`s as in other places.

Reply:
Done in [Emit MathError when casting reward rates to u128](https://github.com/Cardinal-Cryptography/common-amm/pull/86/commits/de86a5c67b68c54924c32d4e236848365cc397c5).